.github/workflows/pin-requirements.yml

```diff
@@ -2,51 +2,75 @@ name: Pin Requirements
 
 on:
   workflow_dispatch:
+    inputs:
+      tag:
+        description: Tag to pin dependencies against.
+        required: true
+        type: string
+
   workflow_call:
+    inputs:
+      tag:
+        description: Tag to pin dependencies against.
+        required: true
+        type: string
 
 permissions:
-  pull-requests: write
-  contents: write
+  contents: read
 
 jobs:
   update-pinned-requirements:
     runs-on: ubuntu-latest
+    env:
+      TMPDIR: ${{ runner.temp }}
+
+    permissions:
+      pull-requests: write # Pull Request creation.
+      contents: write # Branch creation for PR.
+
     steps:
       - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
         with:
           ref: main
 
       - uses: actions/setup-python@d27e3f3d7c64b4bbf8e4abfb9b63b83e846e0435
         with:
-          # NOTE: Update whenever sigstore-python's minimum Python is bumped.
-          python-version: "3.7"
+          python-version-file: install/.python-version
           cache: "pip"
           cache-dependency-path: pyproject.toml
 
       - run: pip install pip-tools
 
-      - name: Determine latest sigstore-python PyPI release
+      - name: Download wheel from GitHub release
         run: |
-          version=$(curl -H "Accept: application/vnd.pypi.simple.v1+json" https://pypi.org/simple/sigstore/ | jq -r '.versions[-1]')
-          echo "sigstore_python_version=$version" >>$GITHUB_ENV
+          release_url="https://api.github.com/repos/sigstore/sigstore-python/releases/tags/${{ inputs.tag }}"
```

Comment:
Nitpick: Propagate `inputs.tag` as a env var (`SIGSTORE_RELEASE_TAG`?) for use here; otherwise an attacker who somehow gets control of `inputs.tag` can do template injection.

Reply:
Let me know if this is what you had in mind! https://github.com/sigstore/sigstore-python/pull/548/files/16185e3ef1306bbe0bb92bf8a8f1c83ffc3aa8a7..2be5cdcea3e7a5386b9149e84833a095245b100c

Reply:
Yep, exactly what I was thinking! Thanks.